packages/web-react/src/components/Button/Button.tsx

```diff
@@ -2,42 +2,40 @@ import React, { ElementType, ForwardedRef, forwardRef } from 'react';
 import classNames from 'classnames';
 import { useStyleProps } from '../../hooks';
 import { SpiritButtonProps } from '../../types';
-import { useButtonAriaProps } from './useButtonAriaProps';
 import { useButtonStyleProps } from './useButtonStyleProps';
+import { useButtonAriaProps } from './useButtonAriaProps';
 
 const defaultProps = {
   color: 'primary',
-  type: 'button',
   isBlock: false,
   isDisabled: false,
   isSquare: false,
-  elementType: 'button',
   size: 'medium',
+  type: 'button',
 };
 
-export const Button = forwardRef<HTMLButtonElement, SpiritButtonProps>(
-  <T extends ElementType = 'button', C = void, S = void>(
-    props: SpiritButtonProps<T, C, S>,
-    ref: ForwardedRef<HTMLButtonElement>,
-  ): JSX.Element => {
-    const { elementType: ElementTag = 'button', children, ...restProps } = props;
-    const { buttonProps } = useButtonAriaProps(props);
-    const { classProps, props: modifiedProps } = useButtonStyleProps(restProps);
-    const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
+const ButtonBase = <T extends ElementType = 'button', C = void, S = void>(
+  props: SpiritButtonProps<T, C, S>,
+  ref: ForwardedRef<HTMLButtonElement>,
+) => {
+  const { elementType: ElementTag = 'button', ...restProps } = props;
+
+  const { buttonProps } = useButtonAriaProps(restProps);
+  const { classProps, props: modifiedProps } = useButtonStyleProps(restProps);
+  const { styleProps, props: otherProps } = useStyleProps(modifiedProps);
+
+  return (
+    <ElementTag
+      {...otherProps}
+      {...buttonProps}
+      ref={ref}
+      className={classNames(classProps, styleProps.className)}
+      style={styleProps.style}
+    />
+  );
+};
 
-    return (
-      <ElementTag
-        {...otherProps}
-        {...styleProps}
-        {...buttonProps}
-        ref={ref}
-        className={classNames(classProps, styleProps.className)}
-      >
-        {children}
-      </ElementTag>
-    );
-  },
-);
+const Button = forwardRef<HTMLButtonElement, SpiritButtonProps<ElementType>>(ButtonBase);
```

Comment:
Why like this? I will prefer to have the same exports as we have before. And also the definition of the component.